cmd/traffic_cache_tool/CacheTool.cc

```diff
@@ -426,6 +432,103 @@ dir_to_offset(const CacheDirEntry *d, const CacheDirEntry *seg)
 #endif
 }
 
+
+CacheDirEntry *
+Stripe::dir_delete_entry(CacheDirEntry *e, CacheDirEntry *p, int s)
+{
+  CacheDirEntry *seg         = dir_segment(s);
+  int no           = dir_next(e);
+  this->_meta[0][0].dirty = 1;
+  if (p) {
+    unsigned int fo = this->freelist[s];
```

Comment:
Some comments might be nice...it's fine for us, now, but we must make this code such that other people have a chance of understanding it. Let's not inflict on others the pain we've had trying to figure out what's going on.